src/boot/efi/vmm.c

```diff
@@ -295,9 +289,14 @@ const char* smbios_find_oem_string(const char *name) {
         left -= type11->header.length;
 
         for (const char *p = s; p < s + left; ) {
-                const char *e = memchr(p, 0, s + left - p);
-                if (!e || e == p) /* Double NUL byte means we've reached the end of the OEM strings. */
+                // Found double NUL
+                if (p[0] == '\0' && p[1] == '\0') {
```

Comment:
same here. it's not ok to dereference two bytes here if the left memory is just 2 bytes.